pycqed/instrument_drivers/physical_instruments/QuTech_VSM_Module.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/python3
+
+"""
+File:       QuTech_VSM_Module.py
+Author:     Jeroen Bergmans, TNO/QuTech
+Purpose:    Instrument driver for Qutech Vector Switch Matrix
+Usage:
+Notes:      It is possible to view the VSM SCPI log using ssh. To do this
+            connect using ssh e.g., "ssh pi@192.168.0.10"
+            Log file can be viewed with "tail -f /var/log/vsm_scpi.log"
+Bugs:       Probably
+"""
+
+import logging
+import socket
+import time
+
+from qcodes import IPInstrument
+from qcodes import validators as vals
+
+
+class QuTechVSMModule(IPInstrument):
+    def __init__(self, name, address, port=5025, **kwargs):
+        super().__init__(name, address, port,
+                         write_confirmation=False,
+                         **kwargs)
+        self.modules = [1, 2, 3, 4, 5, 6, 7, 8]
+        self.channels = [1, 2, 3, 4]
+
+        self.add_parameters()
+
+        # example of how the commands could look
+        self.add_function('reset', call_cmd='*RST')
+
+        # Set socket option NODELAY, this should be handled by the base class.
+        if self._socket:
+            self._socket.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
+
+        self.connect_message()
+
+    def _recv(self):
+        """ Overwrites base IP recv command to ensuring read till EOM
+            Again, this should be handled properly in a base class.
+        """
+        return self._socket.makefile().readline().rstrip()
+
+    def connect_message(self, idn_param='IDN', begin_time=None):
```

Comment:
I think one should not overwrite a method like this for the reason stated. I agree with the general point but I would like to prevent any divergence form the QCoDeS base. As far as I know a proper logging solution is on the list for QCoDeS core.

Reply:
Fixed this by inheriting from SCPI (just like the AWG driver)